### PR TITLE
Improve help text and online documentation for collection download command and improve collection dataset handling

### PR DESCRIPTION
**Pull request by rlc2**
Document what to expect when running the same command again after the download process is interrupted.

Add a new command-line option to allow datasets to be removed and re-downloaded.  Skip existing datasets in the target directory if the option is not present.

**File Changes:**

`FLIR/conservator/cli/managers.py`

- Add help text that appears in response to `conservator collections download --help`
- Add a new option `--overwrite-datasets` to force existing datasets in the target directory for `conservator collections download` to be removed and re-downloaded

`FLIR/conservator/wrappers/collection.py`

- Support the new `--overwrite-datasets` option for `conservator collections download`.  Existing datasets in the target directory will now be skipped if this option is not set, instead of forcing the application to exit with an error

`docs/usage/cli_quickstart.rst`

- Add a paragraph about resuming downloads following the sample command, and include text describing the handling of existing datasets found in the target directory based on the presence or absence of the `--overwrite-datasets` option

[[JIRA Task](https://jiracommercial.flir.com/browse/CON-3261)]

### CODE REVIEW

**Review by nbutler77 — CHANGES_REQUESTED** [review]
As discussed on [slack](https://flir.slack.com/archives/G01QCQVTN0K/p1675787656562299), this PR should also update how datasets are handled; if the dataset exists (i.e. a folder corresponding to the dataset exists in the download location), it should be skipped, unless the user provides an "overwrite" option on the command line, which should be implemented.

File: docs/usage/cli_quickstart.rst [hunk 1]
```diff
@@ -87,6 +87,11 @@ directory::
 
     $ conservator collections download ID -r -d -v -f -m
 
+Due to the potentially large number of files that may be present in a collection, if the
```
Comment (nbutler77) [inline docs/usage/cli_quickstart.rst:90]: Please also describe dataset behavior here.

File: FLIR/conservator/cli/managers.py [hunk 2]
```diff
@@ -163,11 +163,26 @@ def list_media(identifier, recursive):
...
+            corrupted files will be downloaded again.  Datasets will only be
+            re-downloaded if the `--overwrite_datasets` option is supplied.
+            """
         )
         @click.argument("identifier")
         @click.argument("localpath", default=".")
         @click.option("-d", "--datasets", is_flag=True, help="Pull datasets")
+        @click.option(
+            "-o",
+            "--overwrite_datasets",
```
Comment (nbutler77) [inline FLIR/conservator/cli/managers.py:181]: Please use a hyphen instead of an underscore when naming command line options.

**Review by nbutler77 — CHANGES_REQUESTED** [review]

File: docs/usage/cli_quickstart.rst (hunk 1 above)
Comment (rlc2) [reply docs/usage/cli_quickstart.rst:90]: Done.

File: FLIR/conservator/cli/managers.py (hunk 2 above)
Comment (rlc2) [reply FLIR/conservator/cli/managers.py]: Done.

**Review by SJLC — APPROVED** [review]

**Review by nbutler77 — APPROVED** [review]
Functionality looks fine. All tests pass.

**Pull request merged**